Community/Tdarr_Plugin_advo_FFMPEG_NVENC_Tiered_MKV_with_rollback.js

```diff
@@ -0,0 +1,563 @@
+  let videoStream = {};
+  let videoTrack = {};
+
+  const newSize = file.file_size;
+  const oldSize = otherArguments.originalLibraryFile.file_size;
+
+  const ratio = parseInt((newSize / oldSize) * 100, 10);
+
+  const getBound = (bound) => (bound / 100) * oldSize;
+
+  // default values that will be returned
+  const response = {
+    processFile: false,
+    preset: '',
+    container: '.mkv',
+    handBrakeMode: false,
+    FFmpegMode: false,
+    reQueueAfter: true,
+    infoLog: '',
+    maxmux: false,
+  };
+
+  if (newSize > getBound(inputs.upperBound)
+      && file.mediaInfo.track[0].extra !== undefined
+      && file.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE !== undefined
+      && file.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE === 'true'
+      && file.mediaInfo.track[0].extra.ADVOVTRANSCODEDISCARDED === undefined) {
+    if (otherArguments.originalLibraryFile.mediaInfo.track[0].extra !== undefined
+      && otherArguments.originalLibraryFile.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE !== undefined
+      && otherArguments.originalLibraryFile.mediaInfo.track[0].extra.ADVOVTRANSCODEDONE === 'true') {
+      response.processFile = false;
+      response.infoLog += '☑File has already been transcoded! \n';
+      return response;
+    }
+
+    const sizeText = `New file has size ${newSize.toFixed(3)} MB which is ${ratio}% `
+      + `of original file size:  ${oldSize.toFixed(3)} MB`;
+
+    response.infoLog += `${sizeText}. upperBound is ${inputs.upperBound}%. 
+    Discarding transcoded video and using original.`;
+
+    response.processFile = true;
+    response.FFmpegMode = true;
+    response.preset = `<io> -i "${otherArguments.originalLibraryFile.file}" -map 1:v -map 0:a? -map 0:s? `
+      + '-map_metadata 0 -map_chapters 0 -c:v copy -c:a copy -c:s copy -max_muxing_queue_size 9999 '
+      + '-metadata ADVOVTRANSCODEDISCARDED=true ';
+    return response;
+  }
+
+  // find video stream
+  for (let i = 0; i < file.ffProbeData.streams.length; i += 1) {
+    const currStream = file.ffProbeData.streams[i];
+
+    try {
+      if (currStream.codec_type.toLowerCase() === 'video') {
+        if (currStream.codec_name.toLowerCase() !== 'png'
+          && currStream.codec_name.toLowerCase() !== 'bmp'
+          && currStream.codec_name.toLowerCase() !== 'gif'
+          && currStream.codec_name.toLowerCase() !== 'mjpeg') {
+          videoTrackIdx = i;
+          videoStream = currStream;
+          break;
+        }
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
+    }
+  }
+
+  // find video media info track
+  for (let i = 1; i < file.mediaInfo.track.length; i += 1) {
+    const currTrack = file.mediaInfo.track[i];
+
+    try {
+      if (currTrack.StreamOrder !== undefined && parseInt(currTrack.StreamOrder, 10) === videoTrackIdx) {
+        videoTrack = currTrack;
+      }
+    } catch (err) {
+      // eslint-disable-next-line no-console
+      console.log(err);
```

Comment:
Prefer printing errors to the infoLog, since console.logs don't get captured in the job report.

Only going to comment this once, but goes for the whole file.